app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ActiveRecord::Base
+  has_and_belongs_to_many :twitter_users
+  has_and_belongs_to_many :vimeo_users
+
+  validates :name, presence: true
+  validates :uid, presence:true, uniqueness: true
+  validates :provider, presence: true
+
+  def self.find_or_create_from_omniauth(auth_hash)
```

Comment:
nice work on putting this into a model method instead of keeping the logic in the controller :tada: